traits/tests/test_ctraits.py

```diff
@@ -142,6 +142,23 @@ def test_invalid_comparison_mode(self):
         with self.assertRaises(ValueError):
             trait.comparison_mode = 3
 
+    def test_comparison_mode_unchanged_if_invalid(self):
+        trait = CTrait(TraitKind.trait)
+        default_comparison_mode = trait.comparison_mode
+
+        self.assertNotEqual(
+            default_comparison_mode, ComparisonMode.no_compare
+        )
+
+        trait.comparison_mode = ComparisonMode.no_compare
+
+        with self.assertRaises(ValueError):
+            trait.comparison_mode = -1
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.no_compare
+        )
+
     def test_comparison_mode_int(self):
```

Comment:
After #735 and since now `comparison_mode` is being implemented as a property, I think we can have the setter not reset the flags when setting with an invalid value.

Reply:
> have the setter not reset the flags when setting with an invalid value

Yes, that's definitely the expected behaviour!

Reply:
I fixed this and this is ready for another review.